src/main/java/org/broadinstitute/hellbender/engine/spark/JoinReadsWithVariants.java

```diff
@@ -0,0 +1,73 @@
+package org.broadinstitute.hellbender.engine.spark;
+
+import com.google.common.collect.Iterators;
+import htsjdk.variant.variantcontext.VariantContext;
+import org.apache.spark.SparkFiles;
+import org.apache.spark.api.java.JavaPairRDD;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.function.PairFlatMapFunction;
+import org.broadinstitute.hellbender.engine.FeatureDataSource;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.collections.AutoCloseableCollection;
+import org.broadinstitute.hellbender.utils.config.ConfigFactory;
+import org.broadinstitute.hellbender.utils.iterators.CloseAtEndIterator;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.variant.GATKVariant;
+import org.broadinstitute.hellbender.utils.variant.VariantContextVariantAdapter;
+import scala.Tuple2;
+
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Joins an RDD of GATKReads to variant data by copying the variants files to every node, using Spark's file
+ * copying mechanism.
+ */
+public final class JoinReadsWithVariants {
```

Comment:
Moved to `org.broadinstitute.hellbender.utils.spark`. (BTW I put it where it was originally since the name and package echos the other `*JoinReadsWithVariants` classes, however we are going to remove these soon I hope.)